src/main/java/org/jabref/gui/fieldeditors/contextmenu/EditorContextAction.java

```diff
@@ -0,0 +1,95 @@
+        BooleanBinding hasSelectionBinding = Bindings.createBooleanBinding(() -> textInputControl.getSelection().getLength() > 0, textInputControl.selectionProperty());
+        BooleanBinding allSelectedBinding = Bindings.createBooleanBinding(() -> textInputControl.getSelection().getLength() == textInputControl.getLength());
+        BooleanBinding maskTextBinding = (BooleanBinding) BindingsHelper.constantOf(textInputControl instanceof PasswordField); // (maskText("A") != "A");
+
+        if (SHOW_HANDLES) {
+            this.executable.bind(
+                    switch (command) {
+                        case COPY -> editableBinding.and(maskTextBinding.not()).and(hasSelectionBinding);
+                        case CUT -> maskTextBinding.not().and(hasSelectionBinding);
+                        case PASTE -> editableBinding.and(hasStringInClipboardBinding);
+                        case DELETE -> editableBinding.and(hasSelectionBinding);
+                        case SELECT_ALL -> hasTextBinding.and(allSelectedBinding.not());
+                        default -> BindingsHelper.constantOf(true);
+                    });
+        } else {
+            this.executable.bind(
+                    switch (command) {
+                        case COPY -> editableBinding.and(maskTextBinding.not()).and(hasSelectionBinding);
+                        case CUT -> maskTextBinding.not().and(hasSelectionBinding);
+                        case PASTE -> editableBinding.and(hasStringInClipboardBinding);
+                        case DELETE -> editableBinding.and(hasSelectionBinding);
+                        case SELECT_ALL -> hasTextBinding.and(allSelectedBinding.not()); // why was this disabled before?
+                        default -> BindingsHelper.constantOf(true);
+                    });
+        }
+    }
+
+    @Override
+    public void execute() {
+        switch (command) {
+            case COPY -> textInputControl.copy();
+            case CUT -> textInputControl.cut();
+            case PASTE -> textInputControl.paste();
+            case DELETE -> textInputControl.deleteText(textInputControl.getSelection());
+            case SELECT_ALL -> textInputControl.selectAll();
+        }
+        textInputControl.requestFocus();
+    }
+
+    /**
+     * Returns the default context menu items (except undo/redo)
+     */
+    public static List<MenuItem> getDefaultContextMenuItems(TextInputControl textInputControl,
+                                                            KeyBindingRepository keyBindingRepository) {
+        ActionFactory factory = new ActionFactory(keyBindingRepository);
+
+        MenuItem selectAllMenuItem = factory.createMenuItem(StandardActions.SELECT_ALL,
+                new EditorContextAction(StandardActions.SELECT_ALL, textInputControl));
+        if (SHOW_HANDLES) {
+            selectAllMenuItem.getProperties().put("refreshMenu", Boolean.TRUE); // what does that mean?
```

Comment:
What does that property mean? Is this somehow related to the select all option above?

Reply:
Also no idea here. I think, I copied these things from the original JavaFX sources.